pyccel/codegen/printing/cwrappercode.py

```diff
@@ -0,0 +1,351 @@
+        args = ','.join(['{}'.format(self._print(VariableAddress(a)) for a in expr.args])
+        #to change for args rank 1 +
+        if expr.args:
+            code = '{name}("{flags}", {args})'.format(name=name, flags=flags, args=args)
+        else :
+            code = '{name}("")'.format(name=name)
+        return code
+
+    def _print_PyArgKeywords(self, expr):
+        arg_names = ', \n'.join(['"{}"'.format(a) for a in expr.arg_names] + [self._print(Nil())])
+        return ('static char *{name}[] = {{\n'
+                        '{arg_names}\n'
+                        '}};\n'.format(name=expr.name, arg_names = arg_names))
+
+    def _print_FunctionDef(self, expr):
+        # Save all used names
+        used_names = set([a.name for a in expr.arguments] + [r.name for r in expr.results] + [expr.name.name])
+
+        # Find a name for the wrapper function
+        wrapper_name = self.get_new_name(used_names.union(self._global_names), expr.name.name+"_wrapper")
+        self._function_wrapper_names[expr.name] = wrapper_name
+        self._global_names.add(wrapper_name)
+        used_names.add(wrapper_name)
+
+        # Collect local variables
+        wrapper_vars = [a for a in expr.arguments] + [r for r in expr.results]
+        python_func_args = Variable(dtype=PyccelPyObject(),
+                                 name=self.get_new_name(used_names, "args"),
+                                 is_pointer=True)
+        python_func_kwargs = Variable(dtype=PyccelPyObject(),
+                                 name=self.get_new_name(used_names, "kwargs"),
+                                 is_pointer=True)
+        wrapper_args = [Variable(dtype=PyccelPyObject(),
+                                 name=self.get_new_name(used_names, "self"),
+                                 is_pointer=True),
+                        python_func_args, python_func_kwargs]
+        wrapper_results = [Variable(dtype=PyccelPyObject(),
+                                    name=self.get_new_name(used_names, "result"),
+                                    is_pointer=True)]
+
+        arg_names = [a.name for a in expr.arguments]
+        keyword_list_name = self.get_new_name(used_names,'kwlist')
+        keyword_list = PyArgKeywords(keyword_list_name, arg_names)
+
+        wrapper_body = [keyword_list]
+        wrapper_body_translations = []
+
+        parse_args = []
+        # TODO: Simplify (to 1 line?)
+        # TODO (After PR 422): Handle optional args
+        for a in expr.arguments:
+            collect_var, cast_func = self.get_PyArgParseType(used_names, a)
+            if cast_func is not None:
+                wrapper_vars.append(collect_var)
+                cast_func_call = FunctionCall(cast_func, [collect_var])
+                if cast_func.results[0].is_pointer:
+                    wrapper_body_translations.append(AliasAssign(a, cast_func_call))
+                else:
+                    wrapper_body_translations.append(Assign(a, cast_func_call))
```

Comment:
This code is repeated here and for Build. Could `get_PyArgParseType` return the object in `wrapper_body_translations`.

The collection of the additional variables for `wrapper_vars` could be done using set logic:
`wrapper_vars = set(wrapper_vars).union(parse_args)`
although we will have to find a way to preserve the order